File: ToolDemo/app/src/main/java/com/lilanz/tooldemo/API/APIManager.java
package com.lilanz.tooldemo.API;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class APIManager {

    private static final String BASE_PATH = "http://192.168.35.136:14000/";

    private static Retrofit retrofitInstance;

    public static Retrofit getRetrofit() {
        if (retrofitInstance == null) {
            retrofitInstance = new Retrofit.Builder()
                    .baseUrl(BASE_PATH)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }

        return retrofitInstance;
    }

    // 创建类的对象，并返回
    public static <T> T getService(Class<T> clazz) {
        return getRetrofit().create(clazz);
    }

}

File: KotlinTool/app/src/main/java/com/lilanz/kotlintool/api/APIManager.kt
package com.lilanz.kotlintool.api

import com.lilanz.kotlintool.utils.StringUtil
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class APIManager {
    companion object {
        /**
         * 网络请求的基础地址
         */
        const val BASE_PATH: String = "http://192.168.35.136:14000/"          // 样衣拍照
        const val LOGIN_BASE_PATH: String = "http://webt.lilang.com:8901/";   // 是否有登录权限

        private var retrofitMap: HashMap<String?, Retrofit> = HashMap<String?, Retrofit>();
//        private var retrofitInstance: Retrofit? = null;

        fun getRetrofit(path: String?): Retrofit? {
            var tempPath = path;
            if (StringUtil.isEmpty(tempPath)) {
                tempPath = BASE_PATH
            }

            if (!retrofitMap.containsKey(tempPath)) {
                var instance = Retrofit.Builder()
                    .baseUrl(tempPath)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
                retrofitMap.put(tempPath, instance);
            }

            return retrofitMap.get(tempPath)
        }

        fun getService(clazz: Class<*>): Any? {
            return getRetrofit("")?.create(clazz)
        }

        fun getService(path: String?, clazz: Class<*>): Any? {
            return getRetrofit(path)?.create(clazz)
        }
    }
}